src/unix/os390.c

```diff
@@ -258,9 +257,12 @@ int uv_cpu_info(uv_cpu_info_t** cpu_infos, int* count) {
   idx = 0;
   while (idx < *count) {
     cpu_info->speed = *(int*)(info.siv1v2si22v1.si22v1cpucapability);
-    cpu_info->model = uv__malloc(CPCMODEL_LENGTH + 1);
-    memset(cpu_info->model, '\0', CPCMODEL_LENGTH + 1);
-    memcpy(cpu_info->model, info.siv1v2si11v1.si11v1cpcmodel, CPCMODEL_LENGTH);
+    cpu_info->model = uv__malloc(ZOSCPU_MODEL_LENGTH + 1);
+    if (cpu_info->model == NULL) {
+      uv_free_cpu_info(*cpu_infos, idx);
+      return UV_ENOMEM; 
+    }
+    __get_cpu_model(cpu_info->model, ZOSCPU_MODEL_LENGTH + 1);
```

Comment:
Does this zero out the trailing bytes if the model name's length < ZOSCPU_MODEL_LENGTH? Or is it always exactly ZOSCPU_MODEL_LENGTH bytes long?

Reply:
https://www.ibm.com/docs/en/tdmffz/5.7?topic=reference-determine-cpu-serial-number
https://www.ibm.com/docs/en/zos/2.2.0?topic=information-pcca-mapping
https://github.com/ibmruntimes/zoslib/blob/cc10b7c1d6211a2c28b10b540bf406c7148fbf4f/include/zos-sys-info.h#L71

In the PCCA struct, there is a field `PCCACPID` that contains a bitstring of the CPU information.  The documentation states that 4 bytes in this bitstring (bytes 8-11) are dedicated to the CPU model number.  

It is always exactly `ZOSCPU_MODEL_LENGTH` bytes long